components/form-builder/app/shared/LangSwitcher.tsx

```diff
@@ -0,0 +1,70 @@
+import React from "react";
+import { useTranslation } from "next-i18next";
+import { useTemplateStore } from "@components/form-builder/store";
+import { ToggleLeft, ToggleRight } from "@components/form-builder/icons";
+
+interface LangSwitcherProps {
+  descriptionLangKey: string;
+}
+
+export const LangSwitcher = (props: LangSwitcherProps) => {
+  const { descriptionLangKey } = props;
+  const { t } = useTranslation("form-builder");
+  const { toggleTranslationLanguagePriority, translationLanguagePriority } = useTemplateStore(
+    (s) => ({
+      translationLanguagePriority: s.translationLanguagePriority,
+      toggleTranslationLanguagePriority: s.toggleTranslationLanguagePriority,
+    })
+  );
+
+  const switchLang = () => {
+    toggleTranslationLanguagePriority();
+  };
+
+  return (
+    <div className="flex items-baseline">
+      <span
+        id="switch-heading"
+        className="font-bold text-sm whitespace-nowrap mr-2"
+        aria-hidden="true"
+      >
+        {t(descriptionLangKey)}
+      </span>
+      <div
+        role="button spinbutton"
```

Comment:
Because the design does not fit a traditional component like a Switch or Radio or.. the this required some creativity. The main problem this solves is having the button state announce when it changes through aria-activedescendant. The role=button gives the semantics of a button (which this is close to) but a button should not have an aria-activedescendent. To get around this I added a second role that allows this attribute. It's not pretty but it works.

Reply:
I'm might do doing something wrong but I don't hear anything announced when toggling between languages using VoiceOver --- perhaps some other readers pick this up better?